src/main/java/com/esri/samples/localserver/local_server_feature_layer/LocalServerFeatureLayerSample.java

```diff
@@ -0,0 +1,156 @@
+
+      // create a view with a map and basemap
+      map = new ArcGISMap(Basemap.createStreets());
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // track progress of loading feature layer to map
+      featureLayerProgress = new ProgressBar(-100.0);
+      featureLayerProgress.setMaxWidth(APPLICATION_WIDTH / 4);
+
+      // create local server
+      server = LocalServer.INSTANCE;
+      // listen for the status of the local server to change
+      server.addStatusChangedListener(status -> {
+        if (status.getNewStatus() == LocalServerStatus.STARTED) {
+          try {
+            String featureServiceURL =
+                Paths.get(getClass().getResource("/local_server/PointsofInterest.mpk").toURI()).toString();
+            featureService = new LocalFeatureService(featureServiceURL);
+            featureService.addStatusChangedListener(this::addLocalFeatureLayer);
+            featureService.startAsync();
+          } catch (URISyntaxException e) {
+            System.out.println("Failed to find mpk file. " + e.getMessage());
+          }
+        }
+      });
+      server.startAsync();
+
+      // add view to application window
+      stackPane.getChildren().addAll(mapView, featureLayerProgress);
+      StackPane.setAlignment(featureLayerProgress, Pos.CENTER);
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Once the feature service starts, a feature layer is created from that service and added to the map.
+   * <p>
+   * When the feature layer is done loading the view will zoom to the location of were the features were added.
+   * 
+   * @param status status of feature service
+   */
+  private void addLocalFeatureLayer(StatusChangedEvent status) {
+
+    // check that the feature service has started
+    if (status.getNewStatus() == LocalServerStatus.STARTED) {
+      // get the url of where feature service is located
+      String url = featureService.getUrl() + "/0";
+      // create a feature layer using the url
+      ServiceFeatureTable featureTable = new ServiceFeatureTable(url);
+      featureTable.loadAsync();
+      FeatureLayer featureLayer = new FeatureLayer(featureTable);
+      featureLayer.addDoneLoadingListener(() -> {
+        // zoom to location were features were added
+        mapView.setViewpoint(new Viewpoint(featureLayer.getFullExtent().getCenter(), 30000000));
+        // turn off progress, feature layer is loaded
+        Platform.runLater(() -> featureLayerProgress.setVisible(false));
+      });
+      featureLayer.loadAsync();
+      // add feature layer to map
+      map.getOperationalLayers().add(featureLayer);
+
+    }
+  }
+
+  /**
+   * Stops and releases all resources used in application.
+   */
+  @Override
+  public void stop() throws Exception {
+
+    // stop any services and server that is running
+    if (server != null && server.getStatus() == LocalServerStatus.STARTED) {

```

Comment:
Pull this out into a separate method `stopLocalServer` which is called by stop. That way every local server sample has a common method.